doc/raspbian_install.md

```diff
@@ -0,0 +1,130 @@
+
+Note that `pnpm` also needs to do some `npm` stuff, and you'll need to be able to write to a `.npm` directory in $HOME, so let's get that out of the way first:
+
+```sh
+sudo -u goblin bash
+cd /etc/goblin
+corepack use pnpm@8
+```
+
+If you are trying to install to a directory that's on an SD card, go get several cups of coffee and a fun YouTube playlist. You'll be here a while waiting for `corepack use` to finish.
+
+...and then you will find the install fails halfway through! Cypress thinks that you don't satisfy its system requirements. (The Cypress devs do not think people want to use it on a Pi, so their install script doesn't recognize the `armv7` architecture. Their install script defaults to assuming any architecture it doesn't recognize is ia32.) You must fool it. Go into `node_modules/cypress/lib/util.js` and edit the `getRealArch()` function:
+
+```javascript
+async getRealArch() {
+    return 'arm64';
+    // and then the rest of it
+}
+```
+
+Cypress will download a binary that isn't compatible with your Pi, which means you won't be able to run any tests. But you'll at least have succeeded installing pnpm.
+
+## Installing Rust
+
+The `packages/backend/native-utils/` directory contains, well, some native code. It needs `cargo` to be installed. The `cargo` from Raspbian isn't a recent enough version to build with, so you're going to need to install it via the Rust website. We'll install it just for the `goblin` user rather than system-wide for the moment.
+
+```sh
+sudo -u goblin bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+## Building goblin
+
+Time to build!
+
+```sh
+sudo -u goblin bash
+cd /etc/goblin
+source /home/goblin/.cargo/env
+pnpm run build
+```
+
+## Databases
+
+Now let's get the database set up. We're going to go with `goblin` for everything (database name, username, password).
+
+**DO NOT DO THIS IN PRODUCTION**. Even if you _think_ you know your Postgres is nailed down to prevent outside connections. Hope you enjoy your Postgres install housing the metadata for North Korean hacking teams.
+
+```sql
+CREATE USER goblin WITH PASSWORD 'goblin';
+CREATE DATABASE goblin WITH encoding = 'UTF8';
+GRANT ALL ON DATABASE goblin TO goblin;
+```
+
+Now run the migrate step to create the tables (and there's a lot of them):
+
+```sh
+sudo -u goblin bash
+cd /etc/goblin
+pnpm run migrate
+```
+
+## Cleanup
+
+There are a few cleanup steps you should undertake, but we'll skip past them for the moment:
+
+1. Revoke unneeded privs on the `goblin` postgres user. Once you've created the schema, you should only need to update it again when doing migrations. Day-to-day you won't need to `CREATE TABLE`, `ALTER TABLE`, etc.
+2. (...anything else?)
+
+## And now the moment you've been waiting for...
+
+```sh
+sudo -u goblin bash
+cd /etc/goblin
+pnpm run dev
```

Comment:
not in scope for this PR but do we want to / should we also add some rough checklist for production launch?

Given what I've seen so far from the firefish code quality, I'm guessing there could be many vulnerabilities overall, so I wouldn't think too hard about it, but I'd probably add a guide saying "do not have default passwords, do not expose ports you don't need, do not run as root, and then just change config from localhost:3000 to yourdomain.example.com" or so